File: 백준/Gold/1806. 부분합/부분합.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int n = Integer.parseInt(st.nextToken());
        int s = Integer.parseInt(st.nextToken());
        int[] numbers = new int[n];
        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n; i++) {
            numbers[i] = Integer.parseInt(st.nextToken());
        }

        int end = 0;
        int sum = numbers[0];
        int min = Integer.MAX_VALUE;
        for (int start = 0; start < n; start++) {
            while (end < n && sum < s) {
                end++;
                if (end != n) {
                    sum += numbers[end];
                }
            }
            if (end == n) break;
            min = Math.min(min, end - start + 1);
            sum -= numbers[start];
        }
        if (min == Integer.MAX_VALUE) min = 0;

        System.out.print(min);
    }
}

File: 백준/Gold/1806. 부분합/부분합.kt
fun main() {
    val (n, s) = readln().split(" ").map { it.toInt() }
    val list = readln().split(" ").map { it.toInt() }
    var en = 0
    var min = 0x7f7f7f7f
    var sum = list[0]
    for (st in 0 until n) {
        while (en < n && sum < s) {
            en++
            if (en != n) sum += list[en]
        }
        if (en == n) break
        min = Math.min(min, en - st + 1)
        sum -= list[st]
    }
    if (min == 0x7f7f7f7f) min = 0
    print(min)
}